sdk/src/metrics/aggregation/histogram_aggregation.cc

```diff
@@ -13,65 +13,96 @@ namespace metrics
 {
 
 LongHistogramAggregation::LongHistogramAggregation()
-    : boundaries_{0l, 5l, 10l, 25l, 50l, 75l, 100l, 250l, 500l, 1000l},
-      counts_(boundaries_.size() + 1, 0),
-      sum_(0l),
-      count_(0)
+{
+
+  point_data_.boundaries_ = std::list<long>{0l, 5l, 10l, 25l, 50l, 75l, 100l, 250l, 500l, 1000l};
+  point_data_.counts_ =
+      std::vector<uint64_t>(nostd::get<std::list<long>>(point_data_.boundaries_).size() + 1, 0);
+  point_data_.sum_   = 0l;
+  point_data_.count_ = 0;
+}
+
+LongHistogramAggregation::LongHistogramAggregation(HistogramPointData &&data)
+    : point_data_{std::move(data)}
 {}
 
 void LongHistogramAggregation::Aggregate(long value, const PointAttributes &attributes) noexcept
 {
   const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
-  count_ += 1;
-  sum_ += value;
-  for (auto it = boundaries_.begin(); it != boundaries_.end(); ++it)
+  point_data_.count_ += 1;
+  point_data_.sum_ = nostd::get<long>(point_data_.sum_) + value;
+  for (auto it = nostd::get<std::list<long>>(point_data_.boundaries_).begin();
+       it != nostd::get<std::list<long>>(point_data_.boundaries_).end(); ++it)
   {
     if (value < *it)
     {
-      counts_[std::distance(boundaries_.begin(), it)] += 1;
+      point_data_.counts_[std::distance(
```

Comment:
Keep a loop counter which could be used directly to index `point_data_.counts` instead of calling `std::distance`?

Reply:
Good point changed.